### DISCUSSION TITLE
Multi account

### DISCUSSION THREAD

**Discussion by MariaJsn** [sbertix/Swiftagram#140] [Q&A] [answered]
Hi Sbertix :)

Is there any way to support multi account? 
For example: Login with two accounts and when we change it no need to login again.

**Comment by sbertix** [comment, marked as answer]
Hi Maria, 

That said, **Swiftagram** is completely agnostic to account management, as the priority was to support multi account from the start without any possibility of cookies getting messed up, etc. 

All you need to do is fetch a `Secret` for your "new" account (either through `BasicAuthenticator`, `WebViewAuthenticator` or any custom one you might have built for yourself) and pass it around to unlock any endpoint you wish to use. 

`Storage`s, by the default, and clearly I may add, support (virtually) unlimited `Secret`s. 
So, for instance, if you rely on `KeychainStorage` for caching your cookies, `all()` helps you retrieve all `Secret`s. Do you need to retrieve only a specific one for a given user? Just pass the user identifier to any `Storage` `find` method and you'll get it back if it was cached in advance. 
Super easy 😊 

---

<details><summary><strong><I>Disclaimer</i></strong></summary>
<i>This is not a bug, so I'm moving this to the new <strong>Dicussions</strong> section of the repo.</i><br />
<i>Please make sure you're using this instead next time, or the issue will be just deleted without any further explanation.</I>
</details>

**Reply by MariaJsn** [reply]
Unlimited secret is will be good and we use call Storage with parameter to get specific one

**Reply by MariaJsn** [reply]
@sbertix Will you change any code or system is already ready for this problem? :)
If it is ready really I could not understand how can I implement it.

**Reply by sbertix** [reply]
It's already there 😊 

I'm gonna explain again. 
Assuming you're using `KeychainStorage` and you're logged in through `WebViewAuthenticator`, to log in with another account just present `WebViewAuthenticator` once again. 
The new `Secret` will be persisted in `KeychainStorage` so next time you can just retrieve it using `KeychainStorage<Secret>.all()` or, if you've saved the last user identifier in your `UserDefaults` – for instance, `KeychainStorage<Secret>.find(/* identifier of any logged in user */)`. 

To implement it, you basically just log back in and use the different `Secret`s accordingly. 
I don't really understand where your uncertainties come from 🤔

**Reply by MariaJsn** [reply]
You are king thank you so so so much :)

**Reply by huyparody** [reply]
> It's already there 😊
> 
> I'm gonna explain again.
> Assuming you're using `KeychainStorage` and you're logged in through `WebViewAuthenticator`, to log in with another account just present `WebViewAuthenticator` once again.
> The new `Secret` will be persisted in `KeychainStorage` so next time you can just retrieve it using `KeychainStorage<Secret>.all()` or, if you've saved the last user identifier in your `UserDefaults` – for instance, `KeychainStorage<Secret>.find(/* identifier of any logged in user */)`.
> 
> To implement it, you basically just log back in and use the different `Secret`s accordingly.
> I don't really understand where your uncertainties come from 🤔

I can't find the method `KeychainStorage<Secret>.find` anywhere, did you delete it from the newest version?
